rotors_hil_interface/include/rotors_hil_interface/hil_listeners.h

```diff
@@ -0,0 +1,209 @@
+
+    // MAVLINK HIL_GPS message measures GPS velocity in cm/s
+    hil_data->gps_vel = Eigen::Vector3i(ground_speed_msg->twist.linear.x,
+                                        ground_speed_msg->twist.linear.y,
+                                        ground_speed_msg->twist.linear.z) * kMetersToCm;
+
+    hil_data->vel = hil_data->gps_vel.norm();
+  }
+
+  /// \brief Callback for handling IMU messages.
+  /// \param[in] imu_msg An IMU message.
+  /// \param[out] hil_data Pointer to latest data collected for HIL publishing.
+  void ImuCallback(const sensor_msgs::ImuConstPtr& imu_msg,
+                   HilData* hil_data) {
+    boost::mutex::scoped_lock lock(mtx_);
+
+    hil_data->acc = Eigen::Vector3f(imu_msg->linear_acceleration.x,
+                                    imu_msg->linear_acceleration.y,
+                                    imu_msg->linear_acceleration.z);
+
+    hil_data->att = Eigen::Quaterniond(imu_msg->orientation.w,
+                                       imu_msg->orientation.x,
+                                       imu_msg->orientation.y,
+                                       imu_msg->orientation.z);
+
+    hil_data->gyro = Eigen::Vector3f(imu_msg->angular_velocity.x,
+                                     imu_msg->angular_velocity.y,
+                                     imu_msg->angular_velocity.z);
+  }
+
+  /// \brief Callback for handling Magnetometer messages.
+  /// \param[in] mag_msg A Magnetometer message.
+  /// \param[out] hil_data Pointer to latest data collected for HIL publishing.
+  void MagCallback(const sensor_msgs::MagneticFieldConstPtr &mag_msg,
+                   HilData* hil_data) {
+    boost::mutex::scoped_lock lock(mtx_);
+
+    // ROS magnetic field sensor message is in Tesla, while
+    // MAVLINK HIL_SENSOR message measures magnetic field in Gauss.
+    // 1 Tesla = 10000 Gauss
+    hil_data->mag = Eigen::Vector3f(mag_msg->magnetic_field.x,
+                                    mag_msg->magnetic_field.y,
+                                    mag_msg->magnetic_field.z) * kTeslaToGauss;
+  }
+
+  /// \brief Callback for handling Air Pressure messages.
+  /// \param[in] pressure_msg An Air Pressure message.
+  /// \param[out] hil_data Pointer to latest data collected for HIL publishing.
+  void PressureCallback(const sensor_msgs::FluidPressureConstPtr &pressure_msg,
+                        HilData* hil_data) {
+    boost::mutex::scoped_lock lock(mtx_);
+
+    // ROS fluid pressure sensor message is in Pascals, while
+    // MAVLINK HIL_SENSOR message measures fluid pressure in millibar.
+    // 1 Pascal = 0.01 millibar
+    float pressure_mbar = pressure_msg->fluid_pressure * kPascalToMillibar;
+    hil_data->pressure_abs = pressure_mbar;
+
+    // From the following formula: p_stag - p_static = 0.5 * rho * v^2
+    // HIL air speed is in cm/s and is converted to m/s for the purpose of
+    // computing pressure.
+    hil_data->pressure_diff = 0.5 * kAirDensity * hil_data->ind_airspeed *
```

Comment:
It should be initialized as zero but I explicitly added that to the struct constructor